R/use.R

```diff
@@ -98,3 +100,105 @@ copy_if_not_exists <- function(path, new_path, overwrite = FALSE) {
     FALSE
   }
 }
+
+#' Use touchstone GitHub Actions Workflows
+#'
+#' This function will add (or update) the {touchstone} GitHub Actions workflows
+#' to your package repository. Use in the root directory of your repository.
+#' This function will be called by [touchstone::use_touchstone()], you should
+#' only need to call it to update the workflows or change their parameters.
+#' @param overwrite Overwrites files if they exist.
+#' @param on_comment Don't benchmark every commit. Only trigger on PR
+#'  comments starting with `command`.
+#' @param command The command to use to trigger the benchmark with a comment.
+#' @param limit_to Roles that are allowed to trigger the benchmark workflow
+#'   via comment. See details for a list of roles and their definition.
+#'   Set to `NULL` to allow everyone to trigger a benchmark.
+#' @param force_upstream Always benchmark against the upstream base branch.
+#' @return
+#' The function is called for its side effects and returns `NULL` (invisibly).
+#' @details
+#' Possible roles for `limit_to`:
+#' - `OWNER`: Owner of the repository, e.g. user for user/repo.
+#'   - It is undocumented who holds this status in an org.
+#' - `MEMBER`: Member of org for org/repo.
+#' - `COLLABORATOR`: Anyone who was added as a collaborator to a repository.
+#' - `CONTRIBUTOR`: Anyone who has contributed any commit to the repository.
+#'
+#' Each user has only one role and the check does not interpolate permissions,
+#' so you have to add all roles whom you want to have permission to start the
+#' benchmark. So if you only add "COLLABORATOR" the owner will not be able to
+#' start the benchmark.
+#'
+#' GitHub will recognize additional, mostly unusual roles, see the
+#' [documentation](https://docs.github.com/en/rest/issues/comments).
+#' @export
+use_touchstone_workflows <- function(overwrite = FALSE,
+                                     on_comment = FALSE,
+                                     command = "/benchmark",
+                                     limit_to = c("OWNER", "MEMBER", "COLLABORATOR"),
+                                     force_upstream = TRUE) {
+  template <- readLines(
```

Comment:
maybe the file content modification part in `use_touchstone_workflows()` could be factored into another function that we can write a simple unit test for? Like with a reference file and equality comparison?

Reply:
Yeah makes sense, maybe  a snapshot? I looked into validating yaml but that seemed overkill...

Reply:
yeah, if you can do it with a test that snapshot, that's perfect.